actors/power/src/testing.rs

```diff
@@ -209,55 +201,8 @@ fn check_claims_invariants<BS: Blockstore>(
 
     claims_by_address
 }
-fn check_proofs_invariants<BS: Blockstore>(
-    state: &State,
-    store: &BS,
-    claims: &ClaimsByAddress,
-    acc: &MessageAccumulator,
-) -> ProofsByAddress {
-    if state.proof_validation_batch.is_none() {
-        return ProofsByAddress::default();
+fn check_proofs_invariants(state: &State, acc: &MessageAccumulator) {
```

Comment:
what makes use of `check_state_invariants()` across the actors? Is this called by some external tool? I know about the regular go-state-types ones that are run via lotus, but are these run anywhere with any regularity? Or can they be run manually across real state?

Reply:
These are only ever run in actors tests unfortunately.  There used to be one set which was used both in actors tests and live tests of mainnet state.  Ever since we moved to rust actors we've been maintaining twice the code to get the same value.  We've never taken the time to figure out a reasonable solution.

I think it might be slightly less crazy to figure out how to run rust functions on a golang blockstore and then wire this into the lotus-shed command that checks invariants.  Since the FVM must be doing this anyway it can't require completely new code.

Reply:
The integration tests will check invariants across actors.

Yes it's huge technical debt that there are two versions of this. I consider the ones that are developed alongside the actors and run in the tests to be the canonical. The Lotus team, however, consider the version in go-state-types, which runs against mainnet state, to be canonical. I agree with @ZenGround0's suggested path. This isn't captured in an issue anywhere AFAIK, perhaps since it's not quite clear in which repo it belongs.